examples/Bickley_jet_shallow_water.jl

```diff
@@ -0,0 +1,270 @@
+# points `N_z` must be set to one.  Note that ``Lz`` is the mean depth of the fluid. 
+
+Lx = 2π
+Ly = 20
+Lz = 1
+Nx = 128
+Ny = Nx
+
+grid = RegularRectilinearGrid(size = (Nx, Ny, 1),
+                            x = (0, Lx), y = (-Ly/2, Ly/2), z = (0, Lz),
+                            topology = (Periodic, Bounded, Bounded))
+
+# ## Physical parameters
+#
+# This is a toy problem and we choose the parameters so this jet idealizes
+# a relatively narrow mesoscale jet.   
+# The physical parameters are
+#
+#   * ``f``: Coriolis parameter
+#   * ``g``: Acceleration due to gravity
+#   * ``U``: Maximum jet speed
+#   * ``\Delta\eta``: Maximum free-surface deformation that is dictated by geostrophy
+#   * ``\epsilon`` : Amplitude of the perturbation
+
+f = 1
+g = 9.80665
+U = 1.0
+Δη = f * U / g
+ϵ = 1e-4
+nothing # hide
+
+# ## Building a `ShallowWaterModel`
+#
+# We use `grid`, `coriolis` and `gravitational_acceleration` to build the model.
+# Furthermore, we specify `RungeKutta3` for time-stepping and `WENO5` for advection.
+
+model = ShallowWaterModel(
+    timestepper=:RungeKutta3,
+    advection=WENO5(),
+    grid=grid,
+    gravitational_acceleration=g,
+    coriolis=FPlane(f=f),
+    )
+
+# ## Background state and perturbation
+#
+# The background velocity ``\overline u``, free-surface ``\eta`` and vorticity ``\Omega``
+#  are choosen to be a geostrophically balanced Bickely jet with maximum speed of ``U`` and 
+# maximum free-surface deformation of ``Δη``,
+#
+# ```math
+# \overline \eta(y) = - Δη \tanh(y),
+# ```
+# ```math
+# \overline    u(y) =    U \sech^2(y),
+# ```
+# ```math
+# \overilne    Ω(y) =  2 U \sech^2(y) \tanh(y).
+# ```
+#
+# Linear stability theory predicts that for the particular parameters that we consider here,
+# the growth rate for the maximum grwoth rate should be ``0.14``.
+# 
+# We specify `Ω` to be the background vorticity of the jet in the absence of any perturbations.
+# The initial conditions include a small-ampitude perturbation that decays away from the center
+# of the jet.
+
+  Ω(x, y, z) = 2 * U * sech(y)^2 * tanh(y)
+ uⁱ(x, y, z) =   U * sech(y)^2 + ϵ * exp(- y^2 ) * randn()
+ ηⁱ(x, y, z) = -Δη * tanh(y)
+ hⁱ(x, y, z) = model.grid.Lz + ηⁱ(x, y, z)
+uhⁱ(x, y, z) = uⁱ(x, y, z) * hⁱ(x, y, z)
+nothing # hide
+
+# We set the initial conditions for the zonal mass transport `uhⁱ` and height `hⁱ`.
+
+set!(model, uh = uhⁱ , h = hⁱ)
+
+# We compute the total vorticity and the perturbation vorticity.
+
+uh, vh, h = model.solution
+        u = ComputedField(uh / h)
+        v = ComputedField(vh / h)
+        v = ComputedField(v)
+        ω = ComputedField(∂x(vh/h) - ∂y(uh/h))
+   ω_pert = ComputedField(ω - Ω)
+nothing #hide
+
+# ## Running a `Simulation`
+#
+# We pick the time-step so that we make sure we resolve the surface gravity waves, which 
+# propagate with speed of the order ``\sqrt{g L_z}``. That is, with `Δt = 1e-2` we ensure 
+# that `` \sqrt{g L_z} Δt / dx,  \sqrt{g L_z} Δt / dy < 0.7``.
```

Comment:
```suggestion
# that `` \sqrt{g L_z} Δt / Δx,  \sqrt{g L_z} Δt / Δy < 0.7``.
```

(sorry my bad...)

Reply:
That's better thanks for the fix

Reply:
Also, I ran the example locally and agree that it does run so no need for ENV in this code.  Thanks for being persistant!